ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java

```diff
@@ -932,7 +940,9 @@ protected void createBucketForFileIdx(FSPaths fsp, int filesIdx)
             && !FileUtils.mkdir(fs, outPath.getParent(), hconf)) {
           LOG.warn("Unable to create directory with inheritPerms: " + outPath);
         }
-        fsp.outWriters[filesIdx] = HiveFileFormatUtils.getHiveRecordWriter(jc, conf.getTableInfo(),
+        JobConf jobConf = new JobConf(jc);
+        setWriteOperation(jobConf);
+        fsp.outWriters[filesIdx] = HiveFileFormatUtils.getHiveRecordWriter(jobConf, conf.getTableInfo(),
```

Comment:
Why not just add the operation as a parameter to the `getHiveRecordWriter` method? Passing parameters in a conf seems like a bad practice, if not necessary.